src/org/opensolaris/opengrok/web/api/v1/suggester/provider/filter/AuthorizationFilter.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * CDDL HEADER START
+ *
+ * The contents of this file are subject to the terms of the
+ * Common Development and Distribution License (the "License").
+ * You may not use this file except in compliance with the License.
+ *
+ * See LICENSE.txt included in this distribution for the specific
+ * language governing permissions and limitations under the License.
+ *
+ * When distributing Covered Code, include this CDDL HEADER in each
+ * file and include the License file at LICENSE.txt.
+ * If applicable, add the following below this CDDL HEADER, with the
+ * fields enclosed by brackets "[]" replaced with your own identifying
+ * information: Portions Copyright [yyyy] [name of copyright owner]
+ *
+ * CDDL HEADER END
+ */
+
+/*
+ * Copyright (c) 2018 Oracle and/or its affiliates. All rights reserved.
+ */
+package org.opensolaris.opengrok.web.api.v1.suggester.provider.filter;
+
+import org.opensolaris.opengrok.authorization.AuthorizationFramework;
+import org.opensolaris.opengrok.configuration.Project;
+import org.opensolaris.opengrok.configuration.RuntimeEnvironment;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerRequestFilter;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.Provider;
+
+/**
+ * Serves for authorization of specific REST API endpoints.
+ */
+@Provider
+@Authorized
+public class AuthorizationFilter implements ContainerRequestFilter {
+
+    public static final String PROJECTS_PARAM = "projects[]";
```

Comment:
should this be taken from elsewhere ? This literal seems to be specific to the request rather than filter.

Reply:
Yes, but I wanted to make the filter more generic. Now, if some other endpoint takes projects in a param it can be authorized by only using the `@Authorized` annotation.